### PR TITLE
Avoid ParquetDataset API when appending in ArrowDatasetEngine

### PR DESCRIPTION
**Pull request by rjzamora**
This is a small follow-up to [comments](https://github.com/dask/dask/pull/7527/files#r608563661) made in #7527 (Note that I'd like that PR to get in first).  Most of these changes are just moving around existing code.

`ParquetDataset` will likely be deprecated in the nearish future.  This PR ensures that we will avoid using it when the "pyarrow-dataset" engine is specified.

### CODE REVIEW

File: dask/tests/test_distributed.py [hunk 1]
```diff
@@ -549,10 +549,10 @@ async def test_shuffle_priority(c, s, a, b):
     # Make sure most "split" tasks are processing before
     # any "combine" tasks begin
     late_split = np.quantile(
-        [i for i, st in enumerate(log) if st.startswith("split")], 0.75
+        [i for i, st in enumerate(log) if st.startswith("split")], 0.73
     )
     early_combine = np.quantile(
-        [i for i, st in enumerate(log) if st.startswith("simple")], 0.25
+        [i for i, st in enumerate(log) if st.startswith("simple")], 0.27
```
Comment (rjzamora) [inline dask/tests/test_distributed.py:555]: It seems like this test from #7846 has a small chance of randomly failing - This change makes the failure *slightly* less likely to occur.  Perhaps the test should just be marked as flaky, or redesigned?

Comment (rjzamora) [reply dask/tests/test_distributed.py:555]: cc @jrbourbeau (in case you have thoughts)

Comment (jrbourbeau) [reply dask/tests/test_distributed.py:555]: Yeah, this popped up over in https://github.com/dask/dask/pull/7856 (see [this CI build](https://github.com/dask/dask/pull/7856/checks?check_run_id=2974649244)). Just a guess, but could we reduce the partition / `max_branch` size such all the "split" tasks begin processing before any "combine" tasks? If so, that might give us a more deterministic thing to `assert`

Comment (rjzamora) [reply dask/tests/test_distributed.py:555]: Yes - My concern is just that the test may also pass most of the time **without** the "split"-priority change.  I originally chose settings that usually fail without the modified priorities.

Comment (rjzamora) [reply dask/tests/test_distributed.py:555]: Note that I opened #7879 to attack this flaky test seperately.  So, I'll roll back any changes related to the test in this PR.

**Comment by rjzamora** [comment]
This is related to #7871 and should be ready after #7879 goes in

**Comment by jrbourbeau** [comment]
Merged `main` since https://github.com/dask/dask/pull/7879 has been merged

**Review by jrbourbeau — COMMENTED** [review]
Thanks @rjzamora! cc @martindurant @jorisvandenbossche in case they have any thoughts on this. Otherwise I'll plan to merge later today

**Pull request merged**